tests/unit/Entity/ItemTest.php

```diff
@@ -400,4 +401,44 @@ public function testNotEquals( Item $firstItem, Item $secondItem ) {
 		$this->assertFalse( $secondItem->equals( $firstItem ) );
 	}
 
+	public function testClone() {
+		$id = new ItemId( 'Q1' );
+		$fingerprint = new Fingerprint();
+		$fingerprint->setLabel( 'en', 'original' );
+		$siteLink = new SiteLink( 'enwiki', 'Original' );
+		$siteLinks = new SiteLinkList( array( $siteLink ) );
+		$statement = new Statement( new PropertyNoValueSnak( 1 ) );
+		$statements = new StatementList( $statement );
+		$original = new Item( $id, $fingerprint, $siteLinks, $statements );
+
+		$clone = clone $original;
+
+		// The clone must not reference the same mutable objects
+		$this->assertSame( 'item', $clone->getType() );
+		$this->assertSame( $id, $clone->getId(), 'id is immutable and must not be cloned' );
+		$this->assertNotSame( $fingerprint, $clone->getFingerprint() );
+		$this->assertNotSame( $siteLinks, $clone->getSiteLinkList() );
+		$this->assertSame(
+			$siteLink,
+			$clone->getSiteLinkList()->toArray()['enwiki'],
+			'SiteLink is immutable and must not be cloned'
+		);
+		$this->assertNotSame( $statements, $clone->getStatements() );
+		$this->assertNotSame( $statement, $clone->getStatements()->toArray()[0] );
+
+		// Modify the clone
+		$clone->getFingerprint()->setLabel( 'en', 'clone' );
+		$clonedStatement = $clone->getStatements()->toArray()[0];
```

Comment:
Does not work in PHP 5.3 (https://github.com/wmde/WikibaseDataModel/issues/611#issuecomment-182474997)